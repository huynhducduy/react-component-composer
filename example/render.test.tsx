import * as React from 'react';
import * as renderer from 'react-test-renderer';

import composeComponents from '../src';
import TestComponent from './TestComponent';

it('compose 0 component without crashing', () => {
  const Components = composeComponents();

  const tree = renderer.create(<Components>Test</Components>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('compose 1 component without crashing', () => {
  const Components = composeComponents(TestComponent);

  const tree = renderer.create(<Components>Test</Components>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('compose 3 component without crashing', () => {
  const Components = composeComponents(
    TestComponent,
    [TestComponent],
    TestComponent
  );

  const tree = renderer.create(<Components>Test</Components>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('compose html tag component without crashing', () => {
  const Components = composeComponents('div');

  const tree = renderer.create(<Components>Test</Components>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('compose component with props and without crashing', () => {
  const Components = composeComponents(['div', {className: 'testClass'}]);

  const tree = renderer.create(<Components>Test</Components>).toJSON();

  expect(tree).toMatchSnapshot();
});
