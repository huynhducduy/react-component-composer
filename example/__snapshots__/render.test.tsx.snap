// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compose 0 component without crashing 1`] = `"Test"`;

exports[`compose 1 component without crashing 1`] = `
<div>
  Test
</div>
`;

exports[`compose 3 component without crashing 1`] = `
<div>
  <div>
    <div>
      Test
    </div>
  </div>
</div>
`;

exports[`compose component with props and without crashing 1`] = `
<div
  className="testClass"
>
  Test
</div>
`;

exports[`compose html tag component without crashing 1`] = `
<div>
  Test
</div>
`;
